src/blog/2024/01/barcode-scanner-into-nodered.md

```diff
@@ -0,0 +1,42 @@
+---
+title: How to connect a barcode scanner to your Node-RED application
+subtitle: A guide to connecting a barcode scanner to Node-RED.
+description: A guide to connecting a barcode scanner to Node-RED.
+date: 2024-01-24
+authors: ["grey-dziuba"]
+image: /blog/2024/01/images/barcode-scanner.png
+tags:
+    - posts
+    - flowfuse
+    - how-to
+---
+
+Barcode scanners, functioning as Human Interface Devices (HID) similar to keyboards, offer versatile programming options.  Variations of barcode scanners can be seen used from anything from checkout counters, logistics, and to manufacturing erp systems. In our case, we kept it basic and we used one to trigger a Node-RED flow, keeping the process straightforward and efficient.  Don't let that limit your imagination though, with QR codes, you can store just about anything including recipes in a JSON structure.
+
+<!--more-->
+
+## Configuring the scanner and scanning barcodes
+
+We revitalized an older project for this purpose, ensuring it's up-to-date. For Windows users, the setup is straightforward. Start by importing the project, [@gdziuba/node-red-usbhid](https://flows.nodered.org/node/@gdziuba/node-red-usbhid), via the palette manager.  Import these [flows](https://flows.nodered.org/flow/3e08565bc0e024e81325dc028c5da792) to get started.  This initial flow identified as **getHIDdevices** will detect all devices connected to your Node-RED environment. Locate your barcode scanner  in the debug output.  You will see everything from your mouse and keyboard.  If you have just recently added the barcode scanner to your computer, it will probably be found at the end.  Once you find it note its **Product ID** and **Vendor ID**.  For us, they would be identified as 2590 and 32903 respectively.
```

Comment:
```suggestion
We revitalized an older project for this purpose, ensuring it's up-to-date. For Windows users, the setup is straightforward. Start by importing the project, [@gdziuba/node-red-usbhid](https://flows.nodered.org/node/@gdziuba/node-red-usbhid), via the palette manager.  Import these [flows](https://flows.nodered.org/flow/3e08565bc0e024e81325dc028c5da792) to get started.  This initial flow identified as **getHIDdevices** will detect all devices connected to your Node-RED environment. Locate your barcode scanner  in the debug output.  You will see everything from your mouse and keyboard.  If you have just recently added the barcode scanner to your computer, it will probably be found at the end.  Once you find it note its **Product ID** and **Vendor ID**.  For us, they would be identified as 1536 and 1504 respectively.
```

Reply:
Please check if these changes made it to the main branch. I don't think so